lib/src/main/kotlin/org/team1540/rooster/drive/pipeline/PipelineExtensions.kt

```diff
@@ -0,0 +1,17 @@
+@file:JvmName("PipelineExtensions")
+
+package org.team1540.rooster.drive.pipeline
+
+import org.team1540.rooster.util.Executable
+import java.util.function.Consumer
+import java.util.function.Supplier
+
+// somewhat unfortunate names
+operator fun <T> (() -> T).plus(f: (T) -> Unit): () -> Unit = { f(this()) }
+
+operator fun <T> Supplier<T>.plus(f: Consumer<T>): Executable = Executable { f.accept(this.get()) }
+
+@JvmName("plusExecutable")
```

Comment:
Should be renamed to `plusFunction` or something

Reply:
haha yes very good